ckks/encoder.go

```diff
@@ -468,7 +468,18 @@ func (encoder *encoderComplex128) EncodeDiagMatrixAtLvl(level int, diagMatrix ma
 		matrix.Scale = scale
 
 		for i := range diagMatrix {
-			matrix.Vec[i] = encoder.encodeDiagonal(logSlots, level, scale, rotate(diagMatrix[i], i))
+
+			idx := i
+			if idx < 0 {
+				idx += slots
+			}
+			matrix.Vec[idx] = encoder.encodeDiagonal(logSlots, level, scale, diagMatrix[i])
+		}
+
+		if len(diagMatrix) == 1 {
+			matrix.rotOnly = true
```

Comment:
I guess rotOnly is true iff there's only one diagonal and the only diagonal contains 1 everywhere.

Reply:
Yes this is correct, this is how it is defined in the struct. However, I think this is adding complexity without much gain, so I removed this case.